cmd/agent/api/server_ipc.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	gorilla "github.com/gorilla/mux"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+var ipcConfigListener net.Listener
+var allowedConfigPaths = map[string]struct{}{
+	"api_key": {},
+}
+
+func startIPCServer(ipcConfigHostPort string, tlsConfig *tls.Config) (err error) {
+	ipcConfigListener, err = getListener(ipcConfigHostPort)
+	if err != nil {
+		return err
+	}
+
+	configEndpointHandler := func(w http.ResponseWriter, r *http.Request) {
+		vars := gorilla.Vars(r)
+
+		body, err := getConfigMarshalled(vars["path"])
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		_, _ = w.Write(body)
+	}
+
+	configEndpointMux := gorilla.NewRouter()
+	configEndpointMux.HandleFunc("/", configEndpointHandler).Methods("GET")
+	configEndpointMux.HandleFunc("/.", configEndpointHandler).Methods("GET")
+	configEndpointMux.HandleFunc("/{path}", configEndpointHandler).Methods("GET")
+	configEndpointMux.Use(validateToken)
+
+	ipcMux := http.NewServeMux()
+	ipcMux.Handle(
+		"/config/",
+		http.StripPrefix("/config", configEndpointMux))
```

Comment:
I would like to maybe see an API version identifier here for future-proofing (e.g. `/config/v1/`). We could also consider if we can accomodate grpc paths in the future.

Reply:
100% on board with this idea. Having a version prefix as part of the URL would allow us interact with the API in the future with more ease

Reply:
Done